src/coreclr/src/tools/crossgen2/ILCompiler.Reflection.ReadyToRun/ReadyToRunSignature.cs

```diff
@@ -537,6 +554,403 @@ public CorElementType PeekElementType()
+            {
+                return MetadataNameFormatter.FormatHandle(reader, handle);
+            }
+
+            public string GetCanonType()
+            {
+                return "__Canon";
+            }
+
+            public string GetMethodFromMethodDef(MetadataReader reader, MethodDefinitionHandle handle, string owningTypeOverride)
+            {
+                StringBuilder signaturePrefixBuilder = new StringBuilder();
+                uint methodDefToken = (uint)MetadataTokens.GetToken(handle);
+                return MetadataNameFormatter.FormatHandle(
+                    reader,
+                    MetadataTokens.Handle((int)methodDefToken),
+                    namespaceQualified: true,
+                    owningTypeOverride: owningTypeOverride);
+            }
+
+            public string GetMethodFromMemberRef(MetadataReader reader, MemberReferenceHandle handle, string owningTypeOverride)
+            {
+                StringBuilder signaturePrefixBuilder = new StringBuilder();
+                uint methodRefToken = (uint)MetadataTokens.GetToken(handle);
+                return MetadataNameFormatter.FormatHandle(
+                    reader,
+                    MetadataTokens.Handle((int)methodRefToken),
+                    namespaceQualified: true,
+                    owningTypeOverride: owningTypeOverride);
+            }
+
+            public string GetInstantiatedMethod(string uninstantiatedMethod, ImmutableArray<string> instantiation)
+            {
+                StringBuilder builder = new StringBuilder();
+                builder.Append(uninstantiatedMethod);
+                builder.Append("<");
+                for (int typeArgIndex = 0; typeArgIndex < instantiation.Length; typeArgIndex++)
+                {
+                    if (typeArgIndex != 0)
+                    {
+                        builder.Append(", ");
+                    }
+                    builder.Append(instantiation[typeArgIndex]);
+                }
+                builder.Append(">");
+                return builder.ToString();
+            }
+
+            public string GetConstrainedMethod(string method, string constraint)
+            {
+                return $"{method} @ {constraint}";
+            }
+
+            public string GetMethodWithFlags(ReadyToRunMethodSigFlags flags, string method)
+            {
+                StringBuilder builder = new StringBuilder();
+                if ((flags & ReadyToRunMethodSigFlags.READYTORUN_METHOD_SIG_UnboxingStub) != 0)
+                {
+                    builder.Append("[UNBOX] ");
+                }
+                if ((flags & ReadyToRunMethodSigFlags.READYTORUN_METHOD_SIG_InstantiatingStub) != 0)
+                {
+                    builder.Append("[INST] ");
+                }
+                builder.Append(method);
+                return method;
```

Comment:
Why we return the `method` not the `builder`

Reply:
That is a bug, thank for the review